tests/test_s3.py

```diff
@@ -4014,3 +4014,195 @@ def test_s3_access_analyzer_filter_with_no_results(self, test, s3_ownership):
             ]
         }, session_factory=factory)
         test.assertRaises(PolicyExecutionError, p.run)
+
+
+class BucketReplication(BaseTest):
+
+    def test_s3_bucket_replication_filter_no_rule(self):
+        session_factory = self.replay_flight_data("test_s3_replication_rule")
```

Comment:
per other s3 tests, please patch the augment to remove additional fetches not germane to the test. this will reduce the recorded file volume considerably.

Reply:
@kapilt , thanks for the comment, PR is updated.